test/tla/TestSets.tla

```diff
@@ -0,0 +1,252 @@
+    Cardinality({ {}, {2} , {6}, {3}, {2, 6}, {2, 3}, {3, 6}, {2, 6, 3} }) = 8
+
+TestIn ==
+    /\ 1 \in Set1357
+    /\ 3 \in Set1357
+    /\ 5 \in Set1357
+    /\ 7 \in Set1357
+
+TestNotIn ==
+    /\ 0 \notin Set1357
+    /\ 2 \notin Set1357
+    /\ 4 \notin Set1357
+    /\ 6 \notin Set1357
+    /\ 8 \notin Set1357
+
+TestSubseteq ==
+    /\ SetEmpty \subseteq Set123567
+    /\ Set123567 \subseteq Set123567
+    /\ Set26 \subseteq Set263
+    /\ Set263 \subseteq Set263
+    /\ Set1357 \subseteq Set123567
+
+TestNotSubseteq ==
+    /\ ~(Set123567 \subseteq SetEmpty)
+    /\ ~(Set123567 \subseteq Set263)
+    /\ ~(Set263 \subseteq Set26)
+    /\ ~(Set123567 \subseteq Set1357)
+
+TestSetMap ==
+    LET Cards == { Cardinality(S): S \in { Set1357, Set123567, SetEmpty } } IN
+    Cards = { 0, 4, 6 }
+
+TestSetMapSets ==
+    LET Sets == { { i \in S: i < 5 } : S \in { Set1357, Set123567, SetEmpty } } IN
+    Sets = { {}, { 1, 3 }, { 1, 2, 3 } }
+
+\* TODO: FIX IT!
+\* https://github.com/informalsystems/apalache/issues/1358
+TestPowersetEmpty ==
+    LET \* @type: Set(Int);
+        S == {}
+    IN
+    SUBSET S = { {} }
+
+\* TODO: FIX IT!
+\* https://github.com/informalsystems/apalache/issues/1358
+TestPowerset26 ==
+    { {}, { 2 }, { 6 }, { 2, 6 } } = SUBSET Set26
+
+\* TODO: FIX IT!
+\* https://github.com/informalsystems/apalache/issues/1103
+TestPowersetSubsetEq ==
+    (SUBSET Set26) \subseteq SUBSET Set26
+
+\* TODO: FIX IT!
+\* https://github.com/informalsystems/apalache/issues/1339
+TestExistsEmptyInPowerset ==
+    \E S \in SUBSET Set263:
+        S = {}
+
+\* TODO: FIX IT!
+\* https://github.com/informalsystems/apalache/issues/1339
+TestExists15InPowerset ==
+    \E S \in SUBSET Set263:
+        S = { 2, 6 }
+
+TestForallInPowerset ==
+    \A S \in SUBSET Set1357:
+        6 \notin S
+
+\* TODO: FIX IT!
+\* https://github.com/informalsystems/apalache/issues/1360
+TestPowersetCardinality ==
+    Cardinality(SUBSET Set263) = 8
+
+TestUnion ==
+    UNION { Set26, Set1357, SetEmpty } = Set123567
+
+TestUnionEmpty ==
+    LET \* @type: Set(Set(Int));
+        Empty == {}
+    IN
+    UNION Empty = {}
+
+TestForall ==
+    \A i \in Set1357:
+        i = 1 \/ i = 3 \/ i = 5 \/ i = 7
+
+TestExists ==
+    \E i \in Set1357:
+        i = 3
+
+TestChooseMin ==
+    LET min == CHOOSE i \in Set1357:
+        \A j \in Set1357: i <= j
+    IN
+    min = 1
+
+TestChooseSet ==
+    LET S ==
+        CHOOSE T \in { SetEmpty, Set263, Set1357 }:
+            6 \in T
+    IN
+    S = Set263
+
+TestChooseInSingleton ==
+    LET x == CHOOSE i \in { 3 }: TRUE IN
+    x = 3
+
+TestChooseEmpty ==
+    LET x == CHOOSE i \in SetEmpty: TRUE IN
+    x <= 0 \/ x > 0
+
+TestForallSet ==
+    \A S \in { SetEmpty, Set26, Set1357 }:
+        10 \notin S
+
+TestExistsSet ==
+    \E S \in { SetEmpty, Set26, Set1357 }:
+        5 \in S
+
+TestIsFiniteSet ==
+    /\ IsFiniteSet(SetEmpty)
+    /\ IsFiniteSet(Set26)
+    /\ IsFiniteSet(Set1357)
+
+TestIsFiniteSetOnInfinite ==
```

Comment:
This is also commented out below. Can you file and issue and/or link it?

Reply:
Good point! I've forgotten to add the link. Fixed